.github/workflows/main.yml

```diff
@@ -0,0 +1,100 @@
+name: Run App Tests
+
+on:
+  push:
+    branches: [ "main" ]
+  pull_request:
+    branches: [ "main" ]
+
+jobs:
+  component_e2e_testing:
+
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [16.x]
+        python-version: ['3.x' ]
+
+    steps:
+      - uses: actions/checkout@v3
+
+      # set up node
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: 'npm'
+          cache-dependency-path: './electron/ui/package-lock.json'
+
+      # install electron
+      - name: install electron
+        working-directory: ./electron
+        run: npm ci
+
+      # install frontend and test
+      - name: npm install & test
+        working-directory: ./electron/ui
+        run: |
+          npm ci
+          npm test
+
+      # set up python
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: ${{ matrix.python-version }}
+          architecture: x64
+          cache: 'pip'
+          cache-dependency-path: './backend/requirements.txt'  
+
+      # install backend
+      - name: install backend
+        working-directory: backend
+        run: pip install -r requirements.txt
+
+      # clone & install watertap
+      - name: clone & install watertap
+        run: |
+          git clone https://github.com/watertap-org/watertap.git
+          cd watertap
+          git checkout workshop_tutorial
+          pip install .
+      # get idaes extensions
+      - name: get idaes extensions
+        working-directory: backend/app
+        run: idaes get-extensions
+
+      # run backend server
+      - name: run backend server
+        working-directory: backend/app
+        run: uvicorn main:app --reload --host 0.0.0.0 --port 8001 &
```

Comment:
All comments below are "I don't think we should worry much about this now, but good to keep in mind if needed":

- If we start running jobs in parallel, we might need to revisit the IP/ports assignations so that each job's frontend is connecting to the same job's backend only
- If the logs from all the background processes get too mixed up/messy, we might need to redirect both stdout and stderr to separate files